tests/integration/targets/ec2_vpc_route_table/tasks/main.yml

```diff
@@ -9,32 +9,102 @@
+      that:
+      - vpc.vpc.id is defined
+      - vpc.changed
+  - name: Assign IPv6 CIDR block to existing VPC, check mode
+    ec2_vpc_net:
+      cidr_block: 10.228.224.0/21
+      name: '{{ resource_prefix }}_vpc'
+      ipv6_cidr: true
+    check_mode: true
+    register: vpc_update
+  - name: assert that VPC would changed
+    assert:
+      that:
+      - vpc_update.changed
+  - name: Assign Amazon-provided IPv6 CIDR block to existing VPC
+    ec2_vpc_net:
+      cidr_block: 10.228.224.0/21
+      name: '{{ resource_prefix }}_vpc'
+      ipv6_cidr: true
+    register: vpc_update
+  - name: assert that VPC was changed, IPv6 CIDR is configured
+    assert:
+      that:
+      - vpc_update.vpc.id == vpc.vpc.id
+      - vpc_update.changed
+      - vpc_update.vpc.ipv6_cidr_block_association_set | length == 1
+  - name: Fetch existing VPC info
+    ec2_vpc_net_info:
+      filters:
+        "tag:Name": "{{ resource_prefix }}_vpc"
+    register: vpc_info
+  - name: assert vpc net info after configuring IPv6 CIDR
+    assert:
+      that:
+        - vpc_info.vpcs | length == 1
+        - vpc_info.vpcs[0].id == vpc.vpc.id
+        - vpc_info.vpcs[0].ipv6_cidr_block_association_set | length == 1
+        - vpc_info.vpcs[0].ipv6_cidr_block_association_set[0].ipv6_cidr_block_state.state == "associated"
+  - name: get Amazon-provided IPv6 CIDR associated with the VPC
+    set_fact:
+      # Example value: 2600:1f1c:1b3:8f00::/56
+      vpc_ipv6_cidr_block: '{{ vpc_info.vpcs[0].ipv6_cidr_block_association_set[0].ipv6_cidr_block }}'
   - name: create subnets
     ec2_vpc_subnet:
       cidr: '{{ item.cidr }}'
       az: '{{ item.zone }}'
+      assign_instances_ipv6: '{{ item.assign_instances_ipv6 }}'
+      ipv6_cidr: '{{ item.ipv6_cidr  }}'
       vpc_id: '{{ vpc.vpc.id }}'
       state: present
       tags:
         Public: '{{ item.public|string }}'
         Name: "{{ (item.public|bool)|ternary('public', 'private') }}-{{ item.zone }}"
     with_items:
-    - cidr: 10.228.228.0/24
+    - cidr: 10.228.224.0/24
       zone: '{{ availability_zone_a }}'
       public: 'True'
-    - cidr: 10.228.229.0/24
+      assign_instances_ipv6: false
+      ipv6_cidr: null
+    - cidr: 10.228.225.0/24
       zone: '{{ availability_zone_b }}'
       public: 'True'
-    - cidr: 10.228.230.0/24
+      assign_instances_ipv6: false
+      ipv6_cidr: null
+    - cidr: 10.228.226.0/24
+      zone: '{{ availability_zone_a }}'
+      public: 'False'
+      assign_instances_ipv6: false
+      ipv6_cidr: null
+    - cidr: 10.228.227.0/24
+      zone: '{{ availability_zone_b }}'
+      public: 'False'
+      assign_instances_ipv6: false
+      ipv6_cidr: null
+    - cidr: 10.228.228.0/24
       zone: '{{ availability_zone_a }}'
       public: 'False'
-    - cidr: 10.228.231.0/24
+      assign_instances_ipv6: true
+      # Carve first /64 subnet of the Amazon-provided CIDR for the VPC.
+      ipv6_cidr: "{{ vpc_ipv6_cidr_block | ansible.netcommon.ipsubnet(64, 1) }}"
```

Comment:
Note that initially I was trying the test with a /120 prefix length, not a /64.
However, that uncovered https://github.com/ansible-collections/ansible.utils/issues/132 and https://github.com/netaddr/netaddr/issues/241